lib/service/postcard_chat_service.dart

```diff
@@ -0,0 +1,172 @@
+import 'dart:convert';
+import 'dart:io';
+import 'dart:typed_data';
+
+import 'package:autonomy_flutter/common/environment.dart';
+import 'package:autonomy_flutter/common/injector.dart';
+import 'package:autonomy_flutter/model/chat_message.dart' as app;
+import 'package:autonomy_flutter/model/pair.dart';
+import 'package:autonomy_flutter/service/chat_auth_service.dart';
+import 'package:autonomy_flutter/service/tezos_service.dart';
+import 'package:autonomy_flutter/util/log.dart';
+import 'package:crypto/crypto.dart';
+import 'package:eth_sig_util/util/utils.dart';
+import 'package:flutter_chat_types/flutter_chat_types.dart' as types;
+import 'package:libauk_dart/libauk_dart.dart';
+import 'package:web_socket_channel/io.dart';
+import 'package:web_socket_channel/web_socket_channel.dart';
+
+class PostcardChatService {
```

Comment:
consider splitting into 2 classes: an `abstract class` & an `implementation class`

Reply:
Suggested function in the interface:
- `init(...)` -> create the WS Connection if not exist (reuse if already created for the id+add.
- `addListener` -> add a new listener to the service.
- `removeListener` -> remove a listener from the server.
- `dispose` -> close the connection, remove from the maintained list.

Reply:
maybe should add a function to call in the `dispose` of `chat_thread_page` to close the connection if no listeners in the current `ChatService`